sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,514 @@
+
+    - analyze AIA instrument response that tells efficiency of channel using instrument properties
+    - then output a wavelength response
+    - use spectral model (synthetic spectra) - to get ion emissivities as a function of temp/density
+    - obtain temperature response based on chianti spectral contribution functions (emissivity)
+
+AIA ccd's do not provide spectroscopic information, so there is no way to directly apply a wavelength-dependent calibration to the data.
+
+other important variables:
+effective area
+ion emissivity
+electron density
+temperature array
+
+utilize sunpy/instr/tests/test_aia.py
+
+
+    for units use:
+    @u.quantity_input(wave=u.m)
+
+"""
+
+import numpy as np
+
+import astropy.units as u
+import astropy.constants as constants
+import chianti.core as ch
+import chianti.constants as c
+from scipy import integrate
+
+from .aia_read_genx2table import aia_instr_properties_to_table
+import sunpy.data.test as test
+
+
+class Response():
+    """
+    class description here!
+
+    Parameters
+    ----------
+
+    Attributes
+    ----------
+
+    Notes:
+
+    Feedback/ Thoughts are always welcome! -Thanks!
+    """
+
+    def __init__(self, channel_list, **kwargs):  # data_table,
+
+        path_to_genx_dir = kwargs.get('path_to_genx_dir', '')
+        version = kwargs.get('version', 6)
+
+        # If no data table exists, make the data table
+        # if not os.path.exists('channel_properties_' + str(version) + '.csv'):
+        #     aia_read_genx2table.aia_instr_properties_to_table(path_to_genx_dir, channel_list, properties, version, save=True)
+
+        # load in astropy table  ( quicker than reading genx files each time)
+        # self.data_table = Table.read('channel_properties_' + str(version) + '.csv')
+
+        self.data_table = aia_instr_properties_to_table(path_to_genx_dir, channel_list,
+                                                        version, save=True)
+
+        self.channel_list = channel_list
+
+    def get_channel_data(self, channel):
+        """
+        This definition returns a specific row in self.data_table containing all of the channel data.
+        Individual properties of the channel can be obtained by indexing the property.
+        ie:  min_wavelength = get_channel_data(94)['wavemin']
+
+        :param channel: specify which channel of which to obtain data
+        :return: Table row that acts like dictionary
+        """
+
+        if channel in self.data_table['channel']:
+            index = self.channel_list.index(channel)
+            channel_data = self.data_table[index]
+
+            # Return np.array not np.ndarray
+            for n, data in enumerate(channel_data):
+                if type(data) == np.ndarray:
+                    array = []
+                    for i in data:
+                        array.append(i)
+                    channel_data[n] = array
+
+            return channel_data
+
+    def wavelength_range_index(self, channel):
```

Comment:
This function should also set a `wavelength` parameter so that this is accessible as well. Really just a matter of pulling this from the `.genx` file.